core/shared/src/main/scala/fs2/Pull.scala

```diff
@@ -70,83 +210,147 @@ sealed abstract class Pull[+F[_], +O, +R] {
     new Bind[F, O, R, R2](this) {
       def cont(r: Terminal[R]) = r.map(f)
     }
 
-  /** Run `p2` after `this`, regardless of errors during `this`, then reraise any errors encountered during `this`. */
-  def onComplete[F2[x] >: F[x], O2 >: O, R2 >: R](p2: => Pull[F2, O2, R2]): Pull[F2, O2, R2] =
-    handleErrorWith(e => p2 >> Fail(e)) >> p2
+  /** Discards the result type of this pull.
+    *
+    * Alias for `this.map(_ => () )`.
+    *
+    * This is equivalent to `Functor[Pull[F, O, *]].void(this)`
+    */
+  def void: Pull[F, O, Unit] = as(())
 
-  /** If `this` terminates with `Pull.raiseError(e)`, invoke `h(e)`. */
-  def handleErrorWith[F2[x] >: F[x], O2 >: O, R2 >: R](
-      h: Throwable => Pull[F2, O2, R2]
-  ): Pull[F2, O2, R2] =
-    new Bind[F2, O2, R2, R2](this) {
-      def cont(e: Terminal[R2]): Pull[F2, O2, R2] =
-        e match {
-          case Fail(e) =>
-            try h(e)
-            catch { case NonFatal(e) => Fail(e) }
-          case other => other
-        }
-    }
+  /** Alias for `_.map(_ => o2)`.
+    *
+    * This is equivalent to `Functor[Pull[F, O, *]].as(this, constant)`
+    *
+    * @tparam S     The type of the constant,
+    * @param  value the new result value of the pull
+    */
+  def as[S](value: S): Pull[F, O, S] = map(_ => value)
 
-  /** Discards the result type of this pull. */
-  def void: Pull[F, O, Unit] = as(())
 }
 
 object Pull extends PullLowPriority {
 
+  private[this] val unit: Terminal[Unit] = Succeeded(())
+
+  /** The `done` pull is a pure, closed, silent pull. It succeeds by emitting a `Unit`. */
+  val done: Pull[Pure, INothing, Unit] = unit
+
+  /** Lifts a constant result value into a pure silent Pull that succeesfully
+    * returns the given value. This is the `pure` function of the Applicative
+    * instance for Pull.
+    */
+  def pure[F[_], R](r: R): Pull[F, INothing, R] = Succeeded(r)
+
+  /** Lifts the effectful computation `F[R]` into a silent pull in that effect.
+    *
+    * If the `fr` action fails with an error, the lifted pull will fail with
+    * the thrown error.
+    */
+  def eval[F[_], R](fr: F[R]): Pull[F, INothing, R] = Eval[F, R](fr)
+
+  /** Lifts the given output value `O` into a pure, closed, pull that emits the
+    * single output.
+    *
+    * _Note_ using singleton chunks is not memory efficient. If possible,
+    * use the chunk-based `output` method instead.
+    *
+    * @tparam O the type of the output of the generated chunk.
+    * @tparam F the effect type of the pull. This is given as a convenience for
+    *         the API users, much like the type parameter in the `Option.empty`
+    *         method, to avoid closing on `Nothing`. Note, however, that the
+    *         returned pull is _pure_ and performs no effects of any type.
+    */
+  def output1[F[x] >: Pure[x], O](o: O): Pull[F, O, Unit] = Output(Chunk.singleton(o))
+
+  /** Outputs a chunk of values. */
+  def output[O](os: Chunk[O]): Pull[Pure, O, Unit] =
```

Comment:
Why did signature of `output` change here? The original had the `F[x] >: Pure[x]` constraint. Seems odd that we'd support that for `output1` and not `output`.

Reply:
Thouhgt about removing those. IMHO, since higher-kinded type parameters are always a bit of a problem, removing them if they are not needed would improve the usability of the endpoint.

Reply:
If we can do it without any inference issues, I'd be fine with it, though I think we should be consistent. E.g., if we can remove the param and replace with Pure in both output/output1 and everything complies for 2.12/2.13/3, then I'm on board.

Reply:
aren't they needed to avoid `covary`?